### PR TITLE
Command Authorization Exception

### PR DESCRIPTION
**Pull request by modwizcode**
Moving from https://github.com/SpongePowered/SpongeAPI/pull/181 since I'm going to be handling the cleanup. 

This is what I was thinking with the changes I wanted made. @Zidane please take a look at the special case for SimpleCommandService.

### CODE REVIEW

File: src/main/java/org/spongepowered/api/service/command/SimpleCommandService.java [hunk 1]
```diff
@@ -39,10 +39,7 @@
 import org.spongepowered.api.plugin.PluginContainer;
 import org.spongepowered.api.plugin.PluginManager;
 import org.spongepowered.api.service.event.EventManager;
-import org.spongepowered.api.util.command.CommandCallable;
-import org.spongepowered.api.util.command.CommandException;
-import org.spongepowered.api.util.command.CommandMapping;
-import org.spongepowered.api.util.command.CommandSource;
+import org.spongepowered.api.util.command.*;
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/service/command/SimpleCommandService.java]: I thought star imports weren't allowed?

Comment (Minecrell) [inline src/main/java/org/spongepowered/api/service/command/SimpleCommandService.java]: They aren't.

Comment (modwizcode) [inline src/main/java/org/spongepowered/api/service/command/SimpleCommandService.java]: Yeah I've not cleaned the PR up for our guidelines.

File: src/main/java/org/spongepowered/api/util/command/CommandAuthorizationException.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+import javax.annotation.Nullable;
+
+/**
+ * Thrown when a command is called without authorization or the required
+ * permissions.
+ */
+public class CommandAuthorizationException extends CommandException {
+
+    private static final long serialVersionUID = 6013332030938687625L;
```
Comment (sk89q) [inline src/main/java/org/spongepowered/api/util/command/CommandAuthorizationException.java]: Have we actually been implementing these IDs for serialization?

**Comment by sk89q** [comment]
I think this PR is a bit too coupled to permission nodes -- there may be other reasons for lack of authorization.

**Comment by modwizcode** [comment]
I don't see a reason for this to exist anymore.

**Comment by Xemiru** [comment]
If this pushes through then its just making permission checks a little bit more work, having to deal with an exception being thrown instead of just a simple if check.

**Comment by gabizou** [comment]
@modwizcode Closing this as you said there isn't much reason for this to exist. 

As well, it should be noted that (if I remember correctly) @zml2008 is working on a Commands PR for the near future.

**Pull request closed without merging**